kmath-core/src/commonMain/kotlin/space/kscience/kmath/structures/Buffer.kt

```diff
@@ -100,9 +100,29 @@ public fun <T> Buffer<T>.asSequence(): Sequence<T> = Sequence(::iterator)
 public fun <T> Buffer<T>.asIterable(): Iterable<T> = Iterable(::iterator)
 
 /**
- * Converts this [Buffer] to a new [List]
+ * Returns a new [List] containing all elements of this buffer.
  */
-public fun <T> Buffer<T>.toList(): List<T> = asSequence().toList()
+public fun <T> Buffer<T>.toList(): List<T> = when (this) {
+    is ArrayBuffer<T> -> array.toList()
+    is ListBuffer<T> -> list.toList()
+    is MutableListBuffer<T> -> list.toList()
```

Comment:
The same here it is already a list. We do not need to do a protective copy here.

Reply:
I don't want to have broken behavior like: `val a = mutableListOf(1); ListBuffer(a)...; a[0] = 42`

Reply:
We do not have to provide stdlib-level safety guarantees. Performance is more important here.